content/en/blog/2024/otel-collector-anti-patterns/index.md

```diff
@@ -0,0 +1,176 @@
+in that pool fail, one of the other Collectors in the pool can take over. This
+keeps data flowing to your destination sans disruptions. Gateway mode is
+commonly deployed per cluster, data center, or region.
+
+![OTel Collector Agent Mode](otel-collector-gateway.png)
+
+So which should you use? Both agent and gateway.
+
+If you’re collecting telemetry data for your application, place a Collector
+agent alongside your application. If you’re collecting data for infrastructure,
+place a Collector agent alongside your infrastructure. Whatever you do, don’t
+collect telemetry for all of your infrastructure and applications using a single
+Collector. That way, if one Collector fails, the rest of your telemetry
+collection is unaffected.
+
+The telemetry from your Collector agents can then be sent to a Collector
+gateway. Because the gateway sits behind a load balancer, you don’t have a
+single point of failure for exporting telemetry data, typically to your
+observability backend.
+
+_Bottom line:_ Having the right Collector deployment configuration to send data
+to your observability backend ensures higher availability of your telemetry
+collection infrastructure.
+
+### 3- Not monitoring your Collectors
+
+Deploying multiple Collector agents and a Collector gateway is great, but it’s
+not good enough. Wouldn’t it be nice to know when one of your Collectors is
+malfunctioning, or when data is being dropped? That way, you can take action
+before things start to escalate. This is where monitoring your Collectors can be
+very useful.
+
+But how does one monitor a Collector? The OTel Collector already emits
+[metrics for the purposes of its own monitoring](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/monitoring.md).
+These can then be sent to your Observability backend for monitoring.
+
+### 4- Not using the right Collector Distribution (or not building your own distribution)
+
+There are two official distributions of the OpenTelemetry Collector:
+[Core](https://github.com/open-telemetry/opentelemetry-collector), and
+[Contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib).
+
+The Core distribution is a bare-bones distribution of the Collector for OTel
+developers to develop and test. It contains a base set of
+[extensions](/docs/collector/configuration/#service-extensions),
+[connectors](/docs/collector/configuration/#connectors),
+[receivers](/docs/collector/configuration/#receivers),
+[processors](/docs/collector/configuration/#processors), and
+[exporters](/docs/collector/configuration/#exporters).
+
+The Contrib distribution is for non-OTel developers to experiment and learn. It
+also extends the Core distribution, and includes components created by
+third-parties (including vendors and individual community members), that are
+useful to the OpenTelemetry community at large.
+
+Neither Core nor Contrib is meant to be part of your production workload. Core
```

Comment:
I don't think that's right. It's correct that we don't recommend contrib to be used in production, but core is certainly recommended, especially if all users need is a distribution with OTLP receivers and exporters, and a couple of processors in between, like the batch processor.

Reply:
Let me re-word that, because it's not that I'm saying to not use the components here. I'm just saying to not use these as THE distribution.

Reply:
@jpkrohling re-worded it. Let me know what you think!